### DISCUSSION TITLE
PKIX path building failed (Mirth 3.11.0 with OpenJDK-18)

### DISCUSSION THREAD

**Discussion by VineAndBranches** [nextgenhealthcare/connect#5414] [Q&A]
I am able to install a system comprised of Mirth 3.11.0, OpenJDK-15, and SQL Server (mssql-jdbc-9.2.1.jre15).  Everything works fine after the installation process.  However, I have been trying to upgrade to the latest Java version (OpenJDK-18.0.2.1) in the install package but am running into an issue when the Mirth Connect Server attempts to start:

> Failed to initialize pool: The driver could not establish a secure connection to SQL Server by using Secure Sockets Layer (SSL) encryption.  Error: "PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target".

I made sure to also upgrade the SQL Server driver (mssql-jdbc-11.2.0.jre18) as well as copy the options from _.\Mirth Connect\docs\mcservice-java9+.vmoptions_ into the appropriate _vmoptions_ files.

I similarly get this error with OpenJDK-18.0 and OpenJDK-17.0.

I'm not sure if this is strictly related to the openJDK package or something with Mirth.  Searches for this problem seem to indicate I need to add a cert to the Java truststore (not sure which cert that would be...).  Any idea what else to check?  I'm not a big Java user so I could be missing obvious.

**Comment by pacmano1** [comment]
You take a look at https://github.com/nextgenhealthcare/connect/discussions/5087

**Reply by VineAndBranches** [reply]
Thanks for the suggestion @pacmano1 but #5087 did not resolve my issue from what I could tell.  Rather, I discovered that I needed to add the cert that was being used for SQL Server to Java's trust store (e.g., .\jdk-18.0.2.1\lib\security\cacert).  This was not something I needed to do for jdk-15.

However, it seems that when calling **java -jar mirth-cli-launcher.jar _{...}_** I'm still getting errors (see attached), whereas calling **mccommand.exe _{...}_** works for some reason.  Our install package has been using the launcher for some time so I'd prefer to keep it that way, if possible, or at least understand the problem.  Thoughts?  Otherwise, is there an advantage to calling the launcher versus the .exe directly in a batch file?

[mirth-cli-launcher exception.txt](https://github.com/nextgenhealthcare/connect/files/9551717/mirth-cli-launcher.exception.txt)
